pkgs/test/by-name/check-references/Cargo.toml

```diff
@@ -0,0 +1,8 @@
+[package]
+name = "check-references"
```

Comment:
This tool doesn't seem to handle subdirectories (below the directory containing `package.nix`) the way the RFC intends.

For example, I tried running it on `pkgs/development/compilers/gcc` has a subdirectory `common` which contains `.nix` files which refer to `../patches/` at commit 86d0c37bba6b7f8ca20ebd834e793d05be51e40e (this PR) and got a lot of errors:

```
- pkgs/development/compilers/gcc/10/default.nix: File  at line 57 contains the path expression "../fix-bug-80431.patch" which may point outside the directory of that package.
- pkgs/development/compilers/gcc/10/default.nix: File  at line 58 contains the path expression "../11/fix-struct-redefinition-on-glibc-2.36.patch" which may point outside the directory of that package.
- pkgs/development/compilers/gcc/10/default.nix: File  at line 59 contains the path expression "../libstdc++-target.patch" which may point outside the directory of that package.
- pkgs/development/compilers/gcc/10/default.nix: File  at line 60 contains the path expression "../no-sys-dirs.patch" which may point outside the directory of that package.
- pkgs/development/compilers/gcc/10/default.nix: File  at line 61 contains the path expression "../no-sys-dirs-riscv.patch" which may point outside the directory of that package.
...
```

My understanding is that these sorts of references are allowed by the RFC, because these references remain within the `gcc` package directory (which will soon be `pkgs/by-name/gc/gcc/`).

Reply:
I think `check-references` needs to maintain a counter of how many levels below the starting directory it is, and allow up to that many `../`'s.

Reply:
Should not be a problem anymore in the [updated tooling PR](https://github.com/NixOS/nixpkgs/pull/250885).